src/pysdmx/model/message.py

```diff
@@ -34,3 +37,80 @@ class Header(Struct, frozen=True, kw_only=True):
     receiver: Optional[str] = None
     source: Optional[str] = None
     dataset_action: Optional[ActionType] = None
+
+
+ORGS = "OrganisationSchemes"
+CLS = "Codelists"
+CONCEPTS = "ConceptSchemes"
+
+MSG_CONTENT_PKG = {
+    ORGS: ItemScheme,
+    CLS: Codelist,
+    CONCEPTS: ConceptScheme,
+}
+
+
+class Message(Struct, frozen=True):
+    """Message class holds the content of SDMX Message.
+
+    Attributes:
+        content (Dict[str, Any]): Content of the message. The keys are the
+            content type (e.g. ``OrganisationSchemes``, ``Codelists``, etc.),
+            and the values are the content objects (e.g. ``ItemScheme``,
+            ``Codelist``, etc.).
+    """
+
+    content: Dict[str, Any]
+
+    def __post_init__(self) -> None:
+        """Checks if the content is valid."""
+        for content_key, content_value in self.content.items():
+            if content_key not in MSG_CONTENT_PKG:
+                raise ValueError(f"Invalid content type: {content_key}")
```

Comment:
Here as well, in case something is invalid, we could return the ClientError in the pysdmx `errors` module

Reply:
Yes, I will implement it using the ClientError exception.